src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -393,6 +393,9 @@ def get(self, request: Request, project: Project) -> Response:
                         )
                         if codecov_data:
                             result["codecov"] = codecov_data
+                        else:
+                            result["codecov"] = {"status": status.HTTP_204_NO_CONTENT}
+                            raise Exception("Expected data from Codecov but got none.")
```

Comment:
You also need `logger.exception` around `if error.response.status_code != 404:`

We can get this one merged though and you can follow up.